backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,358 @@
+using Omnikeeper.Entity.AttributeValues;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OKPluginVariableRendering
+{
+    public class VariableRendering : CLBBase
+    {
+        private readonly IRelationModel relationModel;
+        private readonly ICIModel ciModel;
+        private readonly ITraitsProvider traitsProvider;
+        private readonly IEffectiveTraitModel effectiveTraitModel;
+        private readonly ILayerModel layerModel;
+        private readonly IAttributeModel attributeModel;
+        public VariableRendering(ICIModel ciModel, ILayerModel layerModel, IEffectiveTraitModel effectiveTraitModel,
+            IRelationModel relationModel, IAttributeModel attributeModel, ILatestLayerChangeModel latestLayerChangeModel,
+            ITraitsProvider traitsProvider) : base(latestLayerChangeModel)
+        {
+            this.relationModel = relationModel;
+            this.ciModel = ciModel;
+            this.effectiveTraitModel = effectiveTraitModel;
+            this.layerModel = layerModel;
+            this.attributeModel = attributeModel;
+            this.traitsProvider = traitsProvider;
+        }
+
+        const int moduleGroupPrio = 10;
+        const int customersGroupPrio = 100;
+        const int networkSegmentGroupPrio = 1000;
+        const int networkInterfaceGroupPrio = 2000;
+        const int assignmentGroupDefaultPrio = 10000;
+        const int ciPrio = 100000;
+
+        public override async Task<bool> Run(Layer targetLayer, JObject config, IChangesetProxy changesetProxy, IModelContext trans, ILogger logger)
+        {
+
+            logger.LogDebug("Start VariableRendering");
+
+            Configuration cfg;
+
+            try
+            {
+                cfg = config.ToObject<Configuration>();
+                logger.LogDebug("Parsed configuration for VariableRendering.");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError("An error ocurred while creating configuration instance.", ex);
+                return false;
+            }
+
+            var layersetVariableRendering = await layerModel.BuildLayerSet(cfg.InputLayerSet.ToArray(), trans);
+
+            var allCIs = await ciModel.GetMergedCIs(new AllCIIDsSelection(), layersetVariableRendering, false, AllAttributeSelection.Instance, trans, changesetProxy.TimeThreshold);
+
+            var activeTrait = await traitsProvider.GetActiveTrait(cfg.BaseCI.RequiredTrait, trans, changesetProxy.TimeThreshold);
+
+            if (activeTrait == null)
+            {
+                logger.LogError($"Could not find baseCI required trait {cfg.BaseCI.RequiredTrait}");
+                return false;
+            }
+
+            var mainCIs = await effectiveTraitModel.FilterCIsWithTrait(allCIs, activeTrait, layersetVariableRendering, trans, changesetProxy.TimeThreshold);
+
+            mainCIs = mainCIs.Take(10);
```

Comment:
I assume you do this for testing purposes, but for merging, please remove that :)

Reply:
Yes this was only for testing purpose. Removed it.